plugin/storage/es/dependencystore/storage.go

```diff
@@ -37,33 +37,43 @@ const (
 
 // DependencyStore handles all queries and insertions to ElasticSearch dependencies
 type DependencyStore struct {
-	ctx         context.Context
-	client      es.Client
-	logger      *zap.Logger
-	indexPrefix string
+	ctx                 context.Context
+	client              es.Client
+	logger              *zap.Logger
+	indexPrefix         string
+	useReadWriteAliases bool
```

Comment:
can be the `indexPrefix` and this flag consolidated into a function that returns an index?

Reply:
Not easily or without rearranging more things.

Since it's a common "Store" with no split between Reader and Writer this variable just carries the general prefix. The decision about the actual index happens in the `WriteDependencies` and `GetDependencies` functions.

Reply:
And if I may add, this index cannot set at creation time when using time-based indices as time moves on and the index name might change with every invocation of `WriteDependencies` and `GetDependencies`